src/toolkit/res_buf.h

```diff
@@ -243,7 +287,7 @@ class ResBuf {
   ///
   /// @throws KeyError the resource object to be pushed is already present
   /// in the buffer.
-  void Push(Resource::Ptr r) {
+  void Push(Resource::Ptr r, bool unpackaged = true) {
```

Comment:
Should this be a property of the `Push` or of the `ResBuf` itself?

Reply:
I'll merge in your new `is_bulk` bool from #1687 and use that

Reply:
I think this is maybe slightly different. I can imagine one mode that forces items to become unpackaged but keeps them discrete, and another that squashes them all into a single unpackaged material. Mine is the latter.

Reply:
Rereading your comment, maybe you were already planning on both...

Reply:
Yeah, i meant doing both!